File: ruby/medium/minimum_size_subarray_sum.rb
# @param {Integer} s
# @param {Integer[]} nums
# @return {Integer}
MAX = ((2 ** 31) -1)
def min_sub_array_len(s, nums)
    min_size = MAX
    i, j, sum = 0, 0, 0
    while j < nums.length
        sum += nums[j]
        while(i <= j && sum >= s)
            min_size = [min_size, (j-i+1)].min
            sum -= nums[i]
            i += 1
        end
        j += 1
    end
    return (min_size == MAX) ? 0 : min_size
end

File: javascript/medium/minimumSizeSubarraySum.js
/**
 * @param {number} s
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(s, nums) {
    let minLen = Number.MAX_SAFE_INTEGER, i = 0, j = 0, sum = 0;
    while(j < nums.length) {
        sum += nums[j];
        while(i <= j && sum >= s) {
            minLen = Math.min(minLen, (j-i+1));
            sum -= nums[i];
            i++;
        }
        j++;
    }
    return (minLen === Number.MAX_SAFE_INTEGER) ? 0 : minLen;
};
